tensorzero_internal/src/model.rs

```diff
@@ -150,8 +142,8 @@ impl ModelConfig {
                 .await;
             match response {
                 Ok(response) => {
-                    let (chunk, stream, raw_request) = response;
```

Comment:
could we peek here? I would like to be able to avoid asking every implementor of provider trait to have to peek and instead make it something the providers that need to do do and then all other providers don't have to worry about